Update urls and made nav responsive

nicoleblom opened the pull request:
### Description of your changes
Renamed dev-staging to dev
Moved landing page into layouts and renamed to index.html
Renamed /content/_index.md to docs.md
Updated urls to navigate to landing page or docs home page where necessary
Made top navigation responsive

Upbound-CLA commented:
[![CLA assistant check](https://cla.upbound.io/pull/badge/not_signed)](https://cla.upbound.io/upbound/docs?pullRequest=79) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.upbound.io/upbound/docs?pullRequest=79) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.upbound.io/check/upbound/docs?pullRequest=79) it.</sub>

plumbis commented:
This looks great! My only comment is I don't think we should go live with the landing page yet. I'd like to wait until we have at least one more section with content (i.e., `/content/knowledge-base`, but we don't have the content yet).

With only `/docs` as a destination from the index it feels like an unnecessary click.

Can we move `/themes/hugo-geekdoc/layouts/index.html` back to `/static` (or just rename it) and move `/content/docs.md` back to `/content/_index.md`?

nicoleblom commented:
Hi Pete, 

I reverted some changes, moving index.html into static and renamed docs.md back to _index.md
Updated links where needed, nav logo and Upbound tag now navigates to {{ .Site.BaseURL }}
Footer logo opens upbound.io in new tab to navigate back to marketing site
